### PR TITLE
Remove prefix remove suffix

### LINKED ISSUE #3299: Add .removeprefix and .removesuffix to Seq object (and subclasses)
Python 3.9 adds these new string methods, which to me would apply equally well to sequence manipulation too - for example removing a start or stop codon, or prefixed barcode/adaptor sequences. There are likely other good motivating examples.

https://docs.python.org/3.9/whatsnew/3.9.html
https://www.python.org/dev/peps/pep-0616/

### PR DESCRIPTION
**Pull request by Benedict-Carling**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #3299 

Adds new python 3.9 str functions to Seq object, removeprefix and remove suffix

New Functionality

```python
>>> from Bio.Seq import Seq
>>> my_seq = Seq("ATGGTGTGTGT")
>>> my_seq
Seq('ATGGTGTGTGT')
>>> my_seq.removeprefix('ATG')
Seq('GTGTGTGT')
```

New functions can also take argument at a Seq Object

```python
>>> from Bio.Seq import Seq
>>> my_seq = Seq("GTGTGTGTTAG")
>>> my_seq
Seq('GTGTGTGTTAG')
>>> stop_codon = Seq("TAG")
>>> my_seq.removesuffix(stop_codon)
Seq('GTGTGTGT')
```

Unit Tests are added to confirm this functionality

Thanks for looking at my first pull request here 👋 im looking forward to contributing more,
Benedict

### CODE REVIEW

**Review by peterjc — CHANGES_REQUESTED** [review]
Thank you for working on this. With some slight changes this looks good (see in line comments). I also think this deserves a mention in the NEWS file.

Note: If #3333 gets merged first, this will need some additional work. I had not anticipated that when I opened #3299 for adding this functionality.

File: Bio/Seq.py [hunk 1]
```diff
@@ -729,6 +729,73 @@ def rstrip(self, chars=None):
...
+            return Seq(str(self).removeprefix(prefix))
+        except AttributeError:
+            if not isinstance(prefix, str):
+                raise TypeError(
+                    f"can't remove object of type {type(prefix)}"
+                    "from Seq try Seq or Str"
+                )
+            # TODO - remove once we require Python 3.9+
+            else:
+                if str(self).startswith(str(prefix)):
```
Comment (peterjc) [inline Bio/Seq.py:760]: You don't need ``str(prefix)`` now, just ``prefix`` is enough.

File: Bio/Seq.py [hunk 2]
```diff
@@ -729,6 +729,73 @@ def rstrip(self, chars=None):
...
+            return Seq(str(self).removesuffix(suffix))
+        except AttributeError:
+            if not isinstance(suffix, str):
+                raise TypeError(
+                    f"can't remove object of type {type(suffix)}"
+                    "from Seq try Seq or Str"
+                )
+            # TODO - remove once we require Python 3.9+
+            else:
+                if str(self).endswith(str(suffix)):
```
Comment (peterjc) [inline Bio/Seq.py:794]: Again, just ``suffix`` rather than ``str(suffix)``

**Comment by peterjc** [comment]
One remaining small tweak aside (remove two now redundant calls to ``str(...)``, see code comments), this looks good to me.

@mdehoon could you review, and can we merge this before #3333?

**Review by peterjc — APPROVED** [review]

**Comment by mdehoon** [comment]
> @mdehoon could you review, and can we merge this before #3333?

This code will change after #3333 gets in, so it makes more sense to do #3333 first.

**Comment by peterjc** [comment]
This will need updating now that bytes are used internally.

**Comment by peterjc** [comment]
@Benedict-Carling do you need help updating this for the recent bytes changes? Or should one of us update the pull request (we'll still credit you of course)?

**Review by peterjc — CHANGES_REQUESTED** [review]
Needs updating for bytes work on the Seq class.

**Comment by peterjc** [comment]
I've rebased this and attempted to update it for the major ``Seq`` objects changes.

**Comment by peterjc** [comment]
TravisCI was failing (some kind of time out), but in any case we've since migrated to GitHub Actions.

Any reason not to rebase this and see how the tests look now?

**Comment by peterjc** [comment]
Reopened, rebased, more tests, and now considers ``UnknownSeq`` explicitly.

**Comment by peterjc** [comment]
CI failed on Linux due to #3834

**Review by peterjc — APPROVED** [review]

**Review by mdehoon — APPROVED** [review]

**Comment by peterjc** [comment]
Thank you both, merging...

**Pull request merged**